docs/grpo.md

```diff
@@ -0,0 +1,85 @@
+<!--
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ -->
+
+# Try GRPO!
+
+This tutorial demonstrates step-by-step instructions for setting up the environment and then training the Llama3.1 8B-IT model on the GSM8K math reasoning benchmark using Group Relative Policy Optimization (GRPO). GRPO can enhance your model's problem-solving skills on mathematical word problems, coding problems, etc.
+
+GRPO is an RL algorithm designed to enhance the reasoning abilities of LLMs. It is a variant of Proximal Policy Optimization (PPO) that reduces memory usage by eliminating the need for a separate value function model. GRPO works by generating multiple responses for a given prompt, evaluating these responses using a reward model, and then calculating a relative advantage based on the group's performance to update the policy.
+
+We use Tunix as the library for GRPO. 
+And we use vLLM as the library for efficient model inference and generation.
+ 
+In this tutorial we use a single host TPUVM such as `v6e-8/v5p-8`. Let's get started!
+
+## Setup your virtual environment
+
+### Create a virtual environment
+
+```
+# Install uv using the standalone installer (doesn't require apt/dpkg)
```

Comment:
Do we want to have these python 3.12 installation instructions here ? setup.sh does all of this